### PR TITLE
Improve request decoding, handle base64 and gzinflate errors

### PR DESCRIPTION
**Pull request by DRvanR**
As per https://github.com/simplesamlphp/saml2/pull/55, seeing this bundle provides its own bindings we should do the same.

### CODE REVIEW

**Comment by DRvanR** [comment]
@rjkip @arothuis

File: src/SAML2/AuthnRequestFactory.php [hunk 1]
```diff
@@ -39,7 +40,27 @@ class AuthnRequestFactory
...
     {
         // the GET parameter is already urldecoded by Symfony, so we should not do it again.
-        $samlRequest = gzinflate(base64_decode($httpRequest->get(AuthnRequest::PARAMETER_REQUEST)));
+        $samlRequest = base64_decode($httpRequest->get(AuthnRequest::PARAMETER_REQUEST), true);
+        if ($samlRequest === false) {
+            throw new InvalidRequestException('Failed decoding the request, did not receive a valid base64 string');
+        }
+
+        // gzinflate throws an error
+        $errorNo = $errorMessage = null;
```
Comment (rjkip) [inline src/SAML2/AuthnRequestFactory.php]: You ill, mate? This is very unlike yourself :stuck_out_tongue: I have no problems with it, though :wink:

The comment can go or maybe be rephrased to "Catch any errors gzinflate triggers" or "gzinflate may trigger an error".

File: src/Tests/SAML2/AuthnRequestFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     * @test
+     * @group                    saml2
+     *
+     * @expectedException \Surfnet\SamlBundle\Http\Exception\InvalidRequestException
+     * @expectedExceptionMessage Failed inflating the request;
+     */
+    public function an_exception_is_thrown_when_a_request_cannot_be_inflated()
+    {
+        // the $ is invalid since it is outside the base64 alphabet and we deserialize in strict mode.
+        $request = new Request([AuthnRequest::PARAMETER_REQUEST => base64_encode('nope, not deflated')]);
```
Comment (rjkip) [inline src/Tests/SAML2/AuthnRequestFactoryTest.php]: Copy-paste comment :wink:

File: src/Tests/SAML2/AuthnRequestFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        // the $ is invalid since it is outside the base64 alphabet and we deserialize in strict mode.
+        $request = new Request([AuthnRequest::PARAMETER_REQUEST => '$']);
+
+        AuthnRequestFactory::createFromHttpRequest($request);
+    }
+
+    /**
+     * @test
+     * @group                    saml2
```
Comment (rjkip) [inline src/Tests/SAML2/AuthnRequestFactoryTest.php]: Please reduce the whitespace here while you're at it :stuck_out_tongue:

**Comment by rjkip** [comment]
Awesome. Some nits, but :+1:

**Comment by arothuis** [comment]
:+1:

**Pull request merged**